gtsfm/evaluation/metrics_report.py

```diff
@@ -125,22 +150,68 @@ def get_figures_for_metrics(metrics_group: GtsfmMetricsGroup) -> Tuple[str, str]
     """
     scalar_metrics = {}
     metrics_dict = metrics_group.get_metrics_as_dict()[metrics_group.name]
-
     # Separate the scalar metrics.
     for metric_name, value in metrics_dict.items():
         if isinstance(value, dict):
             # Metrics with a dict representation must contain a summary.
             if metrics.SUMMARY_KEY not in value:
                 raise ValueError(f"Metric {metric_name} does not contain a summary.")
             # Add a scalar metric for mean of 1D distributions.
-            scalar_metrics["mean_" + metric_name] = value[metrics.SUMMARY_KEY]["mean"]
+            scalar_metrics["median_" + metric_name] = value[metrics.SUMMARY_KEY]["median"]
         else:
             scalar_metrics[metric_name] = value
     table = create_table_for_scalar_metrics(scalar_metrics)
     plots_fig = create_plots_for_distributions(metrics_dict)
     return table, plots_fig
 
 
+def get_figures_for_metrics_and_compare(metrics_group: GtsfmMetricsGroup, metric_path: str) -> Tuple[str, str]:
+    """Gets the tables and plots for individual metrics in a metrics group.
+
+    All scalar metrics are reported in the table.
+    Metrics of 1-D distributions have an entry in the table for the median,
+    and a histogram or box plot as per their property.
+
+    Args:
+        metrics_group: A GtsfmMetricsGroup for any gtsfm module.
+        metric_path: A path to the GTSfM generated metrics (default: gtsfm/result_metrics)
+
+    Returns:
+        A tuple of table and plotly figures as HTML code.
+    """
+    all_metrics_groups = []
+
+    colmap_metric_path = metric_path[: metric_path.rindex("/")] + "/colmap" + metric_path[metric_path.rindex("/") :]
```

Comment:
why not pass this directly to this method?

Reply:
actually you can just pass 2 metrics groups

Reply:
Addressed.